core/object/configuration_info.cpp

```diff
@@ -0,0 +1,88 @@
+/**************************************************************************/
+/*  configuration_info.cpp                                                */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "configuration_info.h"
+
+#ifdef TOOLS_ENABLED
+
+// Registered in editor, to avoid tight coupling.
+void (*ConfigurationInfo::configuration_info_changed_func)(Object *p_object) = nullptr;
+
+ConfigurationInfo ConfigurationInfo::from_variant(const Variant &p_variant) {
+	if (p_variant.get_type() == Variant::STRING) {
+		return ConfigurationInfo(static_cast<String>(p_variant));
+	} else if (p_variant.get_type() == Variant::DICTIONARY) {
+		return ConfigurationInfo(static_cast<Dictionary>(p_variant));
```

Comment:
```suggestion
		return ConfigurationInfo(String(p_variant));
	} else if (p_variant.get_type() == Variant::DICTIONARY) {
		return ConfigurationInfo(Dictionary(p_variant));
```
We don't really use `static_cast`, though not sure if it's the same performance-wise. There is also `p_variant.operator String()` syntax.